lib/checkother.cpp

```diff
@@ -1526,10 +1526,9 @@ void CheckOther::checkConstVariable()
                             callNonConstMethod = true;
                         break;
                     }
-                    if (var->isStlType() && Token::Match(tok, "%var% [")) { // containers whose operator[] is non-const
-                        const Token* typeTok = var->typeStartToken() ? var->typeStartToken()->tokAt(2) : nullptr;
+                    if (var->valueType() && var->valueType()->container && Token::Match(tok, "%var% [")) { // containers whose operator[] is non-const
                         const auto& notConst = CheckClass::stl_containers_not_const;
-                        if (typeTok && notConst.find(typeTok->str()) != notConst.end()) {
+                        if (notConst.find(var->valueType()->container->startPattern) != notConst.end()) {
```

Comment:
There is container->stdAssociativeLike which should work better than string matching.

Reply:
So I have just tried that, and apparently sometimes the `ValueType::container` is not set when we need the type information. `TestClass::const49` is an example.